### PR TITLE
[DT-1081] Fix filter bugs

### PR DESCRIPTION
**Pull request by s-rubenstein**
### Summary

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: cypress/component/DataSearch/dataset_search_filters.spec.js [hunk 1]
```diff
@@ -20,4 +47,23 @@ describe('Data Library Filters', () => {
...
+      datasets[0].participantCount);
+    cy.get('#participantCountMin-range-input').should('have.value',
+      datasets[1].participantCount);
+  });
+
+  it('calls the filter handler on range changes', () => {
+    const filterHandlerStub = cy.stub();
+    const props = { datasets, filters: { ...defaultFilters(datasets), participantCountMin: 2, participantCountMax: 5 }, filterHandler: filterHandlerStub, isFiltered: () => {}};
+    mount(<DatasetFilterList {...props} />);
+    cy.get('#participantCountMax-range-input').type('3').trigger('change');
```
Comment (rushtong) [inline cypress/component/DataSearch/dataset_search_filters.spec.js]: Chaining calls after `type` is [unsafe](https://docs.cypress.io/api/commands/type). A lot of our tests suffer from this pattern, so it is something we should be trying to clean up.

File: src/components/data_search/DatasetSearchTable.jsx [hunk 2]
```diff
@@ -303,7 +293,7 @@ export const DatasetSearchTable = (props) => {
         </Box>
         <Box sx={{display: 'flex', flexDirection: 'row', paddingTop: '2em'}}>
           <Box sx={{width: '14%', padding: '0 1em'}}>
-            <DatasetFilterList datasets={datasets} filterHandler={filterHandler} isFiltered={isFilteredArray} onClear={() => setFilters(defaultFilters)}/>
+            <DatasetFilterList datasets={datasets} filterHandler={filterHandler} filters={filters} isFiltered={isFilteredArray} onClear={() => setFilters(defaultFilters(datasets))}/>
```
Comment (rushtong) [inline src/components/data_search/DatasetSearchTable.jsx:296]: Something about this is breaking the `dataset_search_table.spec.js` tests, but it's not exactly clear what. I think the tests might be a little brittle as well in that they're expecting a populated `response.response.body[0]` object.

Comment (rushtong) [reply src/components/data_search/DatasetSearchTable.jsx:296]: I think I see it. The search filter text is slightly different in this PR which breaks the test. Changing line 68 in `dataset_search_table.spec.js` to:
```
return handler(req, '{"range":{"participantCount":{"gte":100,"lte":"50"}}}');
```
and line 91 to
```
return handler(req, '{"range":{"participantCount":{"gte":100,"lte":100}}}');
```
allows the string matching in the handler function to work correctly.

**Review by rushtong — APPROVED** [review]
Looks reasonable, addresses bugs in ticket. Once tests are passing, 👍🏽

**Review by fboulnois — APPROVED** [review]
looks good, thank you 👍

**Pull request merged**